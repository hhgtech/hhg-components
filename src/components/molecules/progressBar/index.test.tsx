import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { ProgressBar } from '.'
import MockTheme from '../../../misc/mockTheme'

afterEach(cleanup)

const icon = (
  <svg
    data-name="Layer 1"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 19.6 19.6"
  >
    <title>{'Component 3'}</title>
    <path
      d="M7.91 15.8a2.61 2.61 0 00-.11.75 2.14 2.14 0 002 2.25 2.14 2.14 0 002-2.25 2.61 2.61 0 00-.11-.75"
      fill="none"
      stroke="#262626"
      strokeWidth={1.6}
    />
    <path
      d="M15.8 6.8a6 6 0 00-12 0c0 7-3 9-3 9h18s-3-2-3-9z"
      strokeLinecap="round"
      strokeLinejoin="round"
      fill="none"
      stroke="#262626"
      strokeWidth={1.6}
    />
  </svg>
)

test('it renders', () => {
  const progressBar = render(
    <MockTheme>
      <ProgressBar
        type={'step-track'}
        numberOfStep={4}
        labelPrefix="Step"
        detailSteps={[
          {
            text: 'fyjfhj fhhchgchg',
            icon,
            description: 'description',
          },
          {
            text: 'Waiting',
            icon,
            description: 'description',
          },
          {
            text: 'In progress',
            icon,
            description: 'description',
          },
          {
            text: 'Waiting',
            icon,
            description: 'description',
          },
        ]}
        activeStep={1}
        className={'custom-class'}
        onChange={(step) => {
          console.log(step)
        }}
      />
    </MockTheme>,
  )

  const self = progressBar.container.firstChild
  expect(self).toMatchSnapshot()
})
