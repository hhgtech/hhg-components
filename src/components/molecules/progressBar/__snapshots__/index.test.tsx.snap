// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  position: relative;
  width: 100%;
}

.c0 div[role='slider'] {
  -webkit-text-size-adjust: none;
  -webkit-user-select: none;
}

.c0 div[role='slider']:focus {
  outline: none;
}

.c3 {
  position: absolute;
  top: 19px;
  width: 24px;
  height: 24px;
  color: #fff;
  border: 2px solid #fff;
  border-radius: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  background-color: #fff;
  color: #bfbfbf;
  border: 1px solid #bfbfbf;
}

.c3[data-is-active='true'] {
  background-color: #2d87f3;
  color: #fff;
  border: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 36px;
}

.c1[data-track-type='detail'] {
  margin-top: 0;
}

.c1[data-track-type='detail'][data-step-passed='true'] > div {
  border: 1px solid #2d87f3;
  height: 0px;
}

.c1[data-track-type='detail'] > div {
  border: 1px solid #d9d9d9;
  height: 0px;
}

.c2 {
  width: 100%;
  height: 5px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border-radius: 4px;
  background-color: #d9d9d9;
}

.c4 {
  position: absolute;
  top: 32px;
  min-width: 100px;
  text-align: center;
  color: #737373;
  font-size: 13px;
  font-weight: 700;
}

.c4[data-step-passed='true'] {
  color: #262626;
}

<div
  class="c0 custom-class progress-bar"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="0"
    style="left: 0px;"
  >
    1
    <div
      class="c4"
      data-step-passed="true"
    >
      Step 1
    </div>
  </div>
  <div
    class="c3"
    data-is-active="true"
    data-label="1"
    style="left: 0px;"
  >
    2
    <div
      class="c4"
      data-step-passed="true"
    >
      Step 2
    </div>
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="2"
    style="left: 0px;"
  >
    3
    <div
      class="c4"
      data-step-passed="false"
    >
      Step 3
    </div>
  </div>
  <div
    class="c3"
    data-is-active="false"
    data-label="3"
    style="left: 0px;"
  >
    4
    <div
      class="c4"
      data-step-passed="false"
    >
      Step 4
    </div>
  </div>
</div>
`;
